automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/FireCloudClient.scala

```diff
@@ -0,0 +1,113 @@
+package org.broadinstitute.dsde.firecloud.api
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model.headers.{Authorization, OAuth2BearerToken}
+import akka.http.scaladsl.model.{Multipart, _}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, _}
+import akka.util.ByteString
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import org.broadinstitute.dsde.firecloud.config.AuthToken
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutor, Future}
+
+class FireCloudClient {
+  implicit val system = ActorSystem()
+  implicit val materializer = ActorMaterializer()
+  implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+
+  val mapper = new ObjectMapper()
+  mapper.registerModule(DefaultScalaModule)
+
+  // TODO: Is this okay here or should it be in a separate object and explicitly imported?
+  implicit class JsonStringUtil(s: String) {
+    def fromJsonMapAs[A](key: String): Option[A] = parseJsonAsMap.get(key)
+
+    def parseJsonAsMap[A]: Map[String, A] = mapper.readValue(s, classOf[Map[String, A]])
+  }
+
+  private def makeAuthHeader(token: AuthToken): Authorization = {
+    headers.Authorization(OAuth2BearerToken(token.value))
+  }
+
+  private def sendRequest(httpRequest: HttpRequest): HttpResponse = {
+    Await.result(Http().singleRequest(httpRequest), 5.minutes)
+  }
+
+  private def handleResponse(response: HttpResponse): String = {
+    response.status.isSuccess() match {
+      case true =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        Await.result(entityFuture, 100.millis).decodeString("UTF-8")
+      case _ =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        throw new APIException(Await.result(entityFuture, 100.millis).decodeString("UTF-8"))
+    }
+  }
+
+  def parseResponse(response: HttpResponse): String = {
```

Comment:
That is specifically on my list of things to clean up. If I remember correctly, the differences are subtle and have to do with needing to add support for something (I don't remember what) and feeling like it would require minor changes throughout the tests that I wasn't prepared to do just yet. Thanks for the implementation suggestion!